src/cli/options/parse-cli-arguments.js

```diff
@@ -18,7 +19,14 @@ function parseArgv(rawArguments, detailedOptions, logger, keys) {
     argv = pick(argv, keys);
   }
 
-  return normalizeCliOptions(argv, detailedOptions, { logger });
+  const normalized = normalizeCliOptions(argv, detailedOptions, { logger });
+
+  return Object.fromEntries(
+    Object.entries(normalized).map(([key, value]) => {
+      const option = detailedOptions.find(({ name }) => name === key) || {};
+      return [option.forwardToApi || camelCase(key), value];
+    })
+  );
```

Comment:
It looks hard to understand the purpose of the code. Can you add a comment? For example, `Make cli options camelCase for readability`

Reply:
https://github.com/prettier/prettier/pull/11976/commits/45601cff4020fb1adcabf70ce6d26ef95ea79186